src/externs.ts

```diff
@@ -181,6 +181,35 @@ export function generateExterns(
     visitor(stmt, []);
   }
 
+  /**
+   * Convert a qualified name from a .d.ts file or declaration context into a mangled identifier.
+   * E.g. for a qualified name in `export = someName;` or `import = someName;`.
+   * If `someName` is `declare global { namespace someName {...} }`, tsickle must not qualify access
+   * to it with the mangled module namespace as it is emitted in the global namespace. Similarly, if
+   * the symbol is declared in a non-module context, it must not be mangled.
+   */
+  function qualifiedNameToMangledIdentifier(name: ts.Identifier|ts.QualifiedName) {
+    const entityName = getEntityNameText(name);
+    let symbol = typeChecker.getSymbolAtLocation(name);
+    if (symbol) {
+      // If this is an aliased name (e.g. from an import), use the alias to refer to it.
+      if (symbol.flags & ts.SymbolFlags.Alias) {
+        symbol = typeChecker.getAliasedSymbol(symbol);
+      }
+      const alias = mtt.symbolsToAliasedNames.get(symbol);
+      if (alias) return alias;
+      const isGlobalSymbol = symbol && symbol.declarations && symbol.declarations.some(d => {
```

Comment:
isGlobalSymbol looks like an useful general utility fn to extract. I am surprised it doesn't already exist in tsickle.

Reply:
It does feel like that, and I've checked the code base again just now (and did before). We have some code that deals with similar situations in `type_translator.ts`, but its semantics are subtly different. The major difference is that we're emitting into `externs` here, which are necessarily a global namespace, as opposed to `goog.module`.

Reply:
ack.